voc/python/ast.py

```diff
@@ -552,7 +552,121 @@ def visit_If(self, node):
     @node_visitor
     def visit_With(self, node):
         # withitem* items, stmt* body):
-        raise NotImplementedError('No handler for With')
+        #     withitem = (expr context_expr, expr? optional_vars)
+        def opcodes_raise_ifnull(exception, message):
```

Comment:
I see what you're doing here; however, this should be the same effect as an `IF()` pseudo-opcode. Instead of using `*opcodes_raise_ifnull()`, you should be able to use:
```
    IF([], IFNONNULL),
        Java.NEW('org/python/exceptions/AttributeError'),
        JavaOpcodes.LDC_W("__exit__"),
        Java.INIT('org/python/exceptions/AttributeError', 'Ljava/lang/String;')
        JavaOpcodes.ATHROW(),
    END_IF(),
```

Better still would be to add code to voc/python/types/java.py that allow for a simple raising of an exception:
```
    IF([], IFNONNULL),
        Java.THROW(
            'org/python/exceptions/AttributeError', 
            ['Ljava/lang/String;', JavaOpcodes.LDC_W("__exit__")],
        )
    END_IF(),
```